### ISSUE TITLE
Broken functional tests for init role and init scenario

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/molecule#3171]
Few hours ago I discovered that we are only running `unit/` tests on CI/CD and that some of the functional tests we had got broken at some point.

Fixing these would not be easy so I will attempt a gradual fixing, with first priority being to ensure we run all tests, and marking the 4 broken ones accordingly:
```
FAILED src/molecule/test/functional/test_command.py::test_command_init_role[delegated] - assert 1 == 0
FAILED src/molecule/test/functional/test_command.py::test_command_init_scenario[delegated] - assert 1 == 0
FAILED src/molecule/test/functional/test_command.py::test_command_dependency[galaxy] - AssertionError: assert 2 == 0
FAILED src/molecule/test/functional/test_command.py::test_command_dependency[shell]
```

The first two tests are failing because converge fails on the newly created scenarios, for reasons outside our control - like lots of linting errors originating from galaxy templates.

The last two are failing because they utility collection `community.molecule` is installed inside the ephemeral molecule folder and this folder is no longer used by the called commands. That is a regression that was triggered by the introduction of prerun phase, which automatically installs deps into a different location and defines ansible collection include path. When this is defined, it prevents molecule from adding its own ephemeral folder to the include paths. 

While I am not yet sure, I believe it we should likely stop messing with role and collection paths inside molecule itself, basically stop using and configuring `ephemeral/collections` and `ephemeral/roles`. 

I do suspect that the second bug is the same as #3135

**Comment by zhan9san** [comment]
> The last two are failing because they utility collection community.molecule is installed inside the ephemeral molecule folder and this folder is no longer used by the called commands. That is a regression that was triggered by the introduction of prerun phase, which automatically installs deps into a different location and defines ansible collection include path. When this is defined, it prevents molecule from adding its own ephemeral folder to the include paths.

The last two can be fixed by both #3514 and https://github.com/ansible/ansible-compat/pull/135

**Comment by ssbarnea** [comment]
We already have a mechanism for installing required collections at runtime for plugins, we should use the same here.
